# Ploidy Boi's Step 4

## Background
The end goal of almost every genomics pipeline is variant calling, the identification of differences in the sequence, structure, or function either between samples or from a sample to a reference. Variants usually take the form of either single nucleotide variants SNVs (point mutations, insertions, deletions) or structural variants SVs (indels, duplications, inversions, and translocations >50bp). Identification of these variants is usually used in bioinformatic analysis in order to understand patterns and processes in biological systems. For simulation studies such as ours, however, they serve another purpose. By comparing the putatively identified variants with our known simulated variants we can assess the accuracy and effectiveness of bioinformatic tools.

## Tetraploidy
For the tetraploid samples output from Supernova was converted into fasta files with the mkoutput command, using both pseudohaplotype option for pseudassembly and the raw option for raw reads. Reads were then aligned to the reference using BWA, duplicated were marked and variants called with GATK.
Initial fastq outputs from the simulated tetraploid genomes were concerning, varying from 1-20k sequence fragments. This likely indicates highly homologous polyploid genomes can not be assembled de novo with linked read sequence technology. However, reads were successfully mapped to a reference, producing a SAM file. Interestingly, no variants were called from the resulting SAM files. As no variants were simulated from the tetraploid genomes, this result indicates that all reads were successfully mapped back to their original reference point. So, why polyploid linked read sequences may present a challenge to assemble completely de Novo, they may be greatly improved by reference alignment methods.

## SNPs
Variant discovery of simulated single-nucleotide polymorphisms in simulated data can be a fairly straightforward way to verify whether or not your methodology for variant analysis is working. Through utilization of the software, GATK, we were able to analyze how simulation software incorporate randomized SNPs, and whether or not widely-applied variant discovery software is able to identify the total amount of SNPs. Using GATK v4.1, .vcf files were subjected to extraction of SNPs using the softwareâ€™s HaplotypeCaller. SNPs were there selected for further analysis. Variant filtering was carried out based on 7 criteria, (QD < 2.0, QUAL < 30.0, SOR > 3.0, FS > 60.0, MQ < 40.0, MQRankSum < -12.5, and ReadPosRankSum, < -8.0). Output files were analyzed through IGV variant viewer in order to visualize the sites of variation between 4 samples, 0 SNP, 100 SNP, 10k SNP, and 50 SNP. 
Upon failure to discover any variant sites within SNP variable simulated data sets, further analysis is required. Does GATK filter out all of these sites? Why would that happen?

## Duplications and Inversions
[Yan]